qiskit_optimization/translators/docplex_mp.py

```diff
@@ -119,23 +120,172 @@ def to_docplex_mp(quadratic_program: "QuadraticProgram") -> Model:
+        else:
+            for x in left_expr.iter_variables():
+                lin[var_names[x]] = left_expr.get_coef(x)
+
+        if right_expr.is_quad_expr():
+            for x in right_expr.linear_part.iter_variables():
+                lin[var_names[x]] = lin.get(var_names[x], 0.0) - right_expr.linear_part.get_coef(x)
+            for quad_triplet in right_expr.iter_quad_triplets():
+                i = var_names[quad_triplet[0]]
+                j = var_names[quad_triplet[1]]
+                v = quad_triplet[2]
+                quad[i, j] = quad.get((i, j), 0.0) - v
+        else:
+            for x in right_expr.iter_variables():
+                lin[var_names[x]] = lin.get(var_names[x], 0.0) - right_expr.get_coef(x)
+
+        rhs = right_expr.constant - left_expr.constant
+
+        if constraint.sense not in cls._sense_dict:
+            raise QiskitOptimizationError(f"Unsupported constraint sense: {constraint}")
+
+        return lin, quad, cls._sense_dict[constraint.sense], rhs
+
+    @staticmethod
+    def _linear_bounds(var_bounds: Dict[str, Tuple[float, float]], linear: Dict[str, float]):
+        linear_lb = 0.0
+        linear_ub = 0.0
+        for var_name, val in linear.items():
+            x_lb, x_ub = var_bounds[var_name]
+            x_lb *= val
+            x_ub *= val
+            linear_lb += min(x_lb, x_ub)
+            linear_ub += max(x_lb, x_ub)
+        return linear_lb, linear_ub
+
+    @classmethod
+    def _indicator_constraints(
+        cls,
+        var_names: Dict[Var, str],
+        var_bounds: Dict[str, Tuple[float, float]],
+        constraint: IndicatorConstraint,
+        indicator_big_m: Optional[float] = None,
+    ):
+        name = constraint.name
+        binary_var = constraint.binary_var
+        active_value = constraint.active_value
+        linear_constraint = constraint.linear_constraint
+        linear, sense, rhs = _FromDocplexMp._linear_constraint(var_names, linear_constraint)
+        linear_lb, linear_ub = _FromDocplexMp._linear_bounds(var_bounds, linear)
+        if sense == "<=":
+            big_m = indicator_big_m or max(0.0, linear_ub - rhs)
+            if active_value:
+                linear[binary_var.name] = big_m
+                rhs += big_m
+            else:
+                linear[binary_var.name] = -big_m
+            return [(linear, sense, rhs, name)]
+        elif sense == ">=":
+            big_m = indicator_big_m or max(0.0, rhs - linear_lb)
+            if active_value:
+                linear[binary_var.name] = -big_m
+                rhs -= big_m
+            else:
+                linear[binary_var.name] = big_m
+            return [(linear, sense, rhs, name)]
+        elif sense == "==":
+            # for equality constraints, add both GE and LE constraints.
+            # linear2, rhs2, and big_m2 are for the GE constraint.
+            linear2 = linear.copy()
+            rhs2 = rhs
+            big_m = indicator_big_m or max(0.0, linear_ub - rhs)
```

Comment:
If I supply big M as a value of 0, which is not None - then this logic will not use that value the user supplied. Maybe its not a useful value anyway to supply anyway but if it was the intent to compute one when the user supplied any value other than None then this does not quite do that, Referring to its description
```
indicator_big_m: The big-M value for indicator constraints.
    If ``None``, it is automatically derived from the model.
```

Reply:
Thanks. I revised it as follows
```python
big_m = max(0.0, linear_ub - rhs) if indicator_big_m is None else indicator_big_m
```